### PR TITLE
r/aws_elasticsearch_domain: Add LogPublishingOption

### LINKED ISSUE #1953: Feature Request: Elasticsearch slow logs & CloudWatch Logs integration
_This issue was originally opened by @maciejdrozdzowski as hashicorp/terraform#16382. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

As per the announcement: https://aws.amazon.com/blogs/database/viewing-amazon-elasticsearch-service-slow-logs/

### LINKED ISSUE #2280: [feature request] aws_elasticsearch_domain slow logs
AWS added a new feature - slow logs 
The logs are saved into Cloudwatch

The announcement by AWS:
https://aws.amazon.com/about-aws/whats-new/2017/10/optimize-your-amazon-elasticsearch-service-domains-using-slow-logs/

AWS CLI documentation:
http://docs.aws.amazon.com/cli/latest/reference/es/create-elasticsearch-domain.html

### Affected Resource(s)
- aws_elasticsearch_domain

### References
https://aws.amazon.com/about-aws/whats-new/2017/10/optimize-your-amazon-elasticsearch-service-domains-using-slow-logs/
http://docs.aws.amazon.com/cli/latest/reference/es/create-elasticsearch-domain.html

### PR DESCRIPTION
**Pull request by atsushi-ishibashi**
Closes #1953 
Closes #2280

### CODE REVIEW

**Comment by atsushi-ishibashi** [comment]
Require #2337

**Comment by radeksimko** [comment]
Hey @atsushi-ishibashi 
FYI - I just merged https://github.com/terraform-providers/terraform-provider-aws/pull/2243 so there should be no more blockers here. 😺

**Comment by atsushi-ishibashi** [comment]
Ok👍 @radeksimko 
```
TF_ACC=1 go test ./aws -v -run=TestAccAWSElasticSearchDomain_LogPublishingOptions -timeout 120m
=== RUN   TestAccAWSElasticSearchDomain_LogPublishingOptions
--- PASS: TestAccAWSElasticSearchDomain_LogPublishingOptions (868.91s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	868.977s
```

File: aws/resource_aws_elasticsearch_domain.go [hunk 1]
```diff
@@ -169,6 +169,38 @@ func resourceAwsElasticSearchDomain() *schema.Resource {
...
+								for _, str := range validLogTypes {
+									if value == str {
+										return
+									}
+								}
+								errors = append(errors, fmt.Errorf("expected %s to be one of %v, got %s", k, validLogTypes, value))
+								return
+							},
+						},
+						"cloud_watch_logs_log_group_arn": {
```
Comment (radeksimko) [inline aws/resource_aws_elasticsearch_domain.go]: Nitpick: Can you make this `cloudwatch_log_group_arn`?

We use `cloudwatch` (written together) in other resources and [`aws_cloudwatch_log_group`](https://www.terraform.io/docs/providers/aws/r/cloudwatch_log_group.html) is a resource name too. I'm aware we'll drift from the API here, but I think it's 👌  to make an exception as the name they use there is really odd.

**Comment by atsushi-ishibashi** [comment]
Ok👍 @radeksimko

**Pull request merged**